src/main/java/com/solvd/micro9/users/web/controller/UserController.java

```diff
@@ -3,73 +3,76 @@
 import com.solvd.micro9.users.domain.User;
 import com.solvd.micro9.users.service.UserService;
 import com.solvd.micro9.users.web.controller.exception.ServiceIsNotAvailableException;
-import com.solvd.micro9.users.web.controller.openfeign.TicketClient;
 import com.solvd.micro9.users.web.dto.EventDto;
 import com.solvd.micro9.users.web.dto.TicketDto;
 import com.solvd.micro9.users.web.dto.UserDto;
 import com.solvd.micro9.users.web.mapper.UserMapper;
-import feign.FeignException;
 import io.github.resilience4j.circuitbreaker.annotation.CircuitBreaker;
 import lombok.RequiredArgsConstructor;
-import org.springframework.cloud.openfeign.EnableFeignClients;
 import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 import org.springframework.web.client.HttpClientErrorException;
-import org.springframework.web.client.RestTemplate;
-
-import java.util.List;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
 
 @RestController
 @RequestMapping("/api/v1/users")
-@EnableFeignClients
 @RequiredArgsConstructor
 public class UserController {
 
     public static final String TICKET_SERVICE = "TICKET-SERVICE";
     private final UserService userService;
     private final UserMapper userMapper;
-    private final RestTemplate restTemplate;
-    private final TicketClient ticketClient;
+    private final WebClient.Builder webClientBuilder;
     private static final String FIND_EVENTS_BY_USER_ID_URL = "http://" + TICKET_SERVICE + "/api/v1/events/user/";
+    private static final String CREATE_TICKET_URL = "http://" + TICKET_SERVICE + "/api/v1/tickets";
     private static final String USER_SERVICE = "user-service";
 
     @GetMapping
-    public List<UserDto> getAll() {
-        List<User> users = userService.getAll();
+    public Flux<UserDto> getAll() {
+        Flux<User> users = userService.getAll();
         return userMapper.domainToDto(users);
     }
 
     @GetMapping(value = "/{id}")
-    public UserDto findByUserId(@PathVariable(name = "id") Long userId) {
-        User user = userService.findById(userId);
+    public Mono<UserDto> findByUserId(@PathVariable(name = "id") Long userId) {
+        Mono<User> user = userService.findById(userId);
         return userMapper.domainToDto(user);
     }
 
     @GetMapping(value = "/{id}/events")
     @CircuitBreaker(name = USER_SERVICE, fallbackMethod = "circuitBreakerFallbackMethod")
-    public ResponseEntity<EventDto[]> findEventsByUserId(@PathVariable(name = "id") Long userId) {
+    public ResponseEntity<Flux<EventDto>> findEventsByUserId(@PathVariable(name = "id") Long userId) {
```

Comment:
don't mix project reactor with ```ResponseEntity```, since its blocking

Reply:
fix in 3424727